source/base/function_spherical.cc

```diff
@@ -225,7 +225,7 @@ namespace Functions
   template <int dim>
   SymmetricTensor<2,dim>
   Spherical<dim>::hessian (const Point<dim> &p_,
-                           const unsigned int component) const
+                           const unsigned int /* component */) const
```

Comment:
no, i did not add those argument to `svalue()/sgradient()/shessian()`. One probably could pass it through, but this was not my intention and I have not though this case through.